perf(linter/no-extend-native): do not create unnecessary `CompactStr`

overlookmotel opened the pull request:
Small optimization. There's no need to create a `CompactStr` just to search a `Vec<CompactStr>` for a match.

camc314 commented:
### Merge activity

* **Jun 24, 5:57 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jun 24, 5:57 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
perf(linter/no-extend-native): do not create unnecessary `CompactStr` (#11885)

Small optimization. There's no need to create a `CompactStr` just to search a `Vec<CompactStr>` for a match.
